samples/perf_client/perf_client.h

```diff
@@ -431,13 +449,31 @@ namespace client
       size_t& written,
       const std::shared_ptr<RpcTlsClient>& connection)
     {
+      if (write_delay_us > 0) {
```

Comment:
If I'm reading this correctly, we can remove `time_elapsed_between_writes_ns` entirely:

```
while (std::chrono::high_resolution_clock::now() - last_write_time < write_delay)
{
  continue;
}
```

Then we can remove the `if (write_delay > 0)` checks here and below - this while loop will always break on the first iteration, and writing `last_write_time` all the time is harmless.